cmd/checkmetrics/history/history.sh

```diff
@@ -0,0 +1,182 @@
+# Useful when re-setting the checkmetrics baseline data.
+
+set -e
+
+basedir=$(dirname "$0")
+
+# How many recent builds do we evaluate
+NUM_BUILDS=5
+
+# What is the default set of repos (Jenkins jobs) we evaluate
+default_repos=()
+default_repos+=("kata-metrics-runtime-ubuntu-16-04-PR")
+default_repos+=("kata-metrics-tests-ubuntu-16-04-PR")
+repos=()
+
+# What test results do we evaluate for each build
+tests=()
+test_queries=()
+tests+=("boot-times")
+test_queries+=(".\"boot-times\".Results | [.[] | .\"to-workload\".Result] | add / length")
+
+tests+=("memory-footprint")
+test_queries+=(".\"memory-footprint\".Results | .[] | .average.Result")
+
+tests+=("memory-footprint-ksm")
+test_queries+=(".\"memory-footprint-ksm\".Results | .[] | .average.Result")
+
+resultsdir="${basedir}/results"
+indexdir="${basedir}/indexes"
+
+# What is the base URL of the Jenkins server
+urlbase="http://jenkins.katacontainers.io/job"
+
+# Where do we find the recent build number information
+urlindex="api/json"
+
+# Where do we get the actual build results from
+urlartifacts="artifact/go/src/github.com/kata-containers/tests/metrics/results"
+
+# Gather up the results (json) files from all the defined repos for the range
+# of dates?
+gather_data() {
+
+	for repo in "${repos[@]}"; do
+		echo "Getting history for repo $repo"
+		local outpath="${indexdir}/${repo}"
+		local outname="${outpath}/index.json"
+		mkdir -p "${outpath}"
+		local url="${urlbase}/${repo}/${urlindex}"
+		# First, we need the index file for the job so we can get the list of the
+		# last 'n' jobs run.
+		curl -L -o ${outname} $url
+
+		builds=$(jq '.builds | .[] | .number' ${outname} | head -n ${NUM_BUILDS})
+
+		echo "Examining builds: $builds"
+
+		# For each build, for each test, pull down the json results file, if it
+		# exists
+		for build in $builds; do
+			echo "Get results for build $build"
+			local builddir="${resultsdir}/${repo}/${build}"
+			mkdir -p ${builddir}
+			local buildurl="${urlbase}/${repo}/${build}/${urlartifacts}/${testfilename}"
+			echo "Pulling result from $buildurl"
+			for test in "${tests[@]}"; do
+				local testfile=${builddir}/${test}.json
+				local testurl="${buildurl}/${test}.json"
+				echo "    $testurl"
+				# Can fail if the build failed to generate any results
+				curl -L -o ${testfile} $testurl || true
+			done
+		done
+	done
+}
+
+# For each test type, process all the relevant data files in the results subdir.
+# *NOTE*, this does *not* take into account the number or list of build numbers we
+# pulled down - it will evaluate all files it finds. If you want to only evaluate
+# the data you pulled, ensure the result directory is empty (or non-existant) before
+# you run the script.
+process_data() {
+	local count=0
+	for test in "${tests[@]}"; do
+		query="${test_queries[$count]}"
+		echo "Processing $test"
+		echo " Query '$query'"
+		count=$((count+1))
+
+		local allvalues
+		local found=0
+		local total=0
+		local min=99999999
```

Comment:
Arf ;)

Reply:
Not wanting to think the Linux shell is in any way lacking, I had a think and came up with...

```sh
$ max=$(printf "%u" -1)
echo $max
18446744073709551615
```

:smile:

Reply:
heh .... done!